app/uk/gov/hmrc/gform/fileupload/FileUploadConnector.scala

```diff
@@ -31,6 +32,13 @@ class FileUploadConnector(wSHttp: WSHttp, baseUrl: String) {
     wSHttp.GET[Envelope](s"$baseUrl/envelopes/${envelopeId.value}")
   }
 
+  def getMaybeEnvelope(envelopeId: EnvelopeId)(implicit hc: HeaderCarrier): Future[Option[Envelope]] = {
```

Comment:
What is justification of keeping `getEnvelope` method around? 

It looks like we should use `getMaybeEnvelope` exclusively.

Reply:
getMaybeEnvelope is used only when entering at new-form, it is used to determine whether there is an envelope to go with an existing form:

if no form then create envelope & form
if form but no envelope then delete form and create new envelope and form
if for and envelope then offer the choice to continue with existing form

Note that no new interaction design is required to do that.

getEnvelope is used at all other times.  If ever the envelope is not present then we simply fail, it can only happen is the envelope expires during the user's session.  To handle that better we would need to have interactions design to explain what had happened to the user, not only when getting the envelope but when adding to it.

Reply:
> To handle that better we would need to have interactions design to explain what had happened to the user, not only when getting the envelope but when adding to it.

That is exactly my concern. Are we going to address this?

Reply:
@SeanMurray1 can we implement `getMaybeEnvelope` in terms of `getEnvelope`?

Reply:
> > To handle that better we would need to have interactions design to explain what had happened to the user, not only when getting the envelope but when adding to it.
> 
> That is exactly my concern. Are we going to address this?

We don't have a story for that, we could.

But the existing story title is clear.  The story did ask for the user to told at new-form time that the envelope had expired, but we removed that in part because it would require interaction design

Reply:
> @SeanMurray1 can we implement `getMaybeEnvelope` in terms of `getEnvelope`?

@SeanMurray1 yes, do that.  Just DRY